tools/icu/icutrim.py

```diff
@@ -285,14 +285,14 @@ def addTreeByType(tree, mytree):
         # read in the resource list for the tree
         treelistfile = os.path.join(options.tmpdir,"%s.lst" % tree)
         runcmd("iculslocs", "-i %s -N %s -T %s -l > %s" % (outfile, dataname, tree, treelistfile))
-        fi = open(treelistfile, 'rb')
-        treeitems = fi.readlines()
-        trees[tree]["locs"] = [treeitems[i].strip() for i in range(len(treeitems))]
-        fi.close()
-        if tree not in config.get("trees", {}):
-            print(" Warning: filter file %s does not mention trees.%s - will be kept as-is" % (options.filterfile, tree))
-        else:
-            queueForRemoval(tree)
+        with io.open(treelistfile, 'r', encoding='utf-8') as fi:
+            treeitems = fi.readlines()
+            trees[tree]["locs"] = [treeitems[i].strip() for i in range(len(treeitems))]
+            fi.close()
```

Comment:
Lines 290-295 should all be de-dented and line 291 can be deleted because the `with open() as fi:` _context manager_ which will auto-close the `fi` at the end of that block.

Reply:
Thanks @cclauss , I agree I made a mistake. it is fixed now